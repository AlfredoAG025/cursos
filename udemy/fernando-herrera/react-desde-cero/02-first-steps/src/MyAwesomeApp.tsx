import type { CSSProperties } from "react";

const firstName = 'Freddy';
const lastName = 'Arroyo';

const favoriteGames = ['Hollow Knight', 'Metroid'];
const isActive = false;

const address = {
    zipCode: 'ABC-123',
    country: 'Canada',
}

const myStyles: CSSProperties = {
    backgroundColor: '#bababa',
    borderRadius: 20,
    padding: 10,
    marginTop: 30,
}
export const MyAwesomeApp = () => {
    return (
        <div data-testid="div-app">
            <h1 data-testid="first-name-title">{firstName}</h1>
            <h3>{lastName}</h3>

            <p className="mi-clase-favorita">{favoriteGames.join(', ')}</p>

            <h1>{isActive ? 'Activo' : 'No Activo'}</h1>

            <p style={myStyles}>{JSON.stringify(address)}</p>
        </div>
    )
}