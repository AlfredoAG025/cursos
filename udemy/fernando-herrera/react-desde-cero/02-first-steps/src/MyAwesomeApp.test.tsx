import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";

import { render, screen } from "@testing-library/react";

describe('MyAwesomeApp', () => {
    test('should render first name & last name', () => {
        const { container } = render(<MyAwesomeApp />);

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');

        expect(h1?.innerHTML).toContain('Freddy');
        expect(h3?.innerHTML).toContain('Arroyo');
    });

    test('should render first name & last name - screen', () => {
        render(<MyAwesomeApp />);
        // screen.debug();

        // const h1 = screen.getByRole('heading', {level: 1});
        const h1 = screen.getByTestId('first-name-title');
        expect(h1?.innerHTML).toContain('Freddy');
    });

    test('should match snapshot', () => {
        const {container} = render(<MyAwesomeApp />);
        expect(container).toMatchSnapshot();
    });

    test('should match snapshot - screen', () => {
        render(<MyAwesomeApp />);
        expect(screen.getByTestId('div-app')).toMatchSnapshot();
        screen.debug();
    });
});