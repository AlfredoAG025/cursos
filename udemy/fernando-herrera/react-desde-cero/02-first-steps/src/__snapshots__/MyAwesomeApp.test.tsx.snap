// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot - screen 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Freddy
  </h1>
  <h3>
    Arroyo
  </h3>
  <p
    class="mi-clase-favorita"
  >
    Hollow Knight, Metroid
  </p>
  <h1>
    No Activo
  </h1>
  <p
    style="background-color: rgb(186, 186, 186); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"ABC-123","country":"Canada"}
  </p>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Freddy
    </h1>
    <h3>
      Arroyo
    </h3>
    <p
      class="mi-clase-favorita"
    >
      Hollow Knight, Metroid
    </p>
    <h1>
      No Activo
    </h1>
    <p
      style="background-color: rgb(186, 186, 186); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC-123","country":"Canada"}
    </p>
  </div>
</div>
`;
